### ISSUE TITLE
checkpoint ts is not correctly forwarded after some rebalance or leader change

### ISSUE THREAD

**Issue by amyangfei** [pingcap/ticdc#391]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

some asynchronously kv read in etcd maybe fail because of disorder

2. What did you expect to see?

3. What did you see instead?

> [2020/03/27 03:08:42.012 +00:00] [ERROR] [scheduler.go:70] ["error on running processor"] [captureid=dd1d353c-6d54-40ea-b26e-ed3652c0dcbd] [changefeedid=285fec9d-8e8c-4dde-a7d2-145901b4ded7] [processorid=d9c7d9fc-db9b-4682-bcf5-d34862d3a4fb] [error="failed to update info: changefeed: 285fec9d-8e8c-4dde-a7d2-145901b4ded7, capture: dd1d353c-6d54-40ea-b26e-ed3652c0dcbd: task status not exists"] [errorVerbose="task status not exists\nchangefeed: 285fec9d-8e8c-4dde-a7d2-145901b4ded7, capture: dd1d353c-6d54-40ea-b26e-ed3652c0dcbd\ngithub.com/pingcap/ticdc/cdc/kv.CDCEtcdClient.GetTaskStatus\n\tgithub.com/pingcap/ticdc@/cdc/kv/etcd.go:317\ngithub.com/pingcap/ticdc/cdc/roles/storage.(*ProcessorTsEtcdRWriter).UpdateInfo\n\tgithub.com/pingcap/ticdc@/cdc/roles/storage/etcd.go:93\ngithub.com/pingcap/ticdc/cdc.(*processor).updateInfo\n\tgithub.com/pingcap/ticdc@/cdc/processor.go:354\ngithub.com/pingcap/ticdc/cdc.(*processor).positionWorker.func1.1\n\tgithub.com/pingcap/ticdc@/cdc/processor.go:272\ngithub.com/pingcap/ticdc/pkg/retry.Run.func1\n\tgithub.com/pingcap/ticdc@/pkg/retry/retry.go:31\ngithub.com/cenkalti/backoff.RetryNotify\n\tgithub.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:37\ngithub.com/cenkalti/backoff.Retry\n\tgithub.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:24\ngithub.com/pingcap/ticdc/pkg/retry.Run\n\tgithub.com/pingcap/ticdc@/pkg/retry/retry.go:30\ngithub.com/pingcap/ticdc/cdc.(*processor).positionWorker.func1\n\tgithub.com/pingcap/ticdc@/cdc/processor.go:271\ngithub.com/pingcap/ticdc/cdc.(*processor).positionWorker\n\tgithub.com/pingcap/ticdc@/cdc/processor.go:340\ngithub.com/pingcap/ticdc/cdc.(*processor).Run.func1\n\tgithub.com/pingcap/ticdc@/cdc/processor.go:201\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:57\nruntime.goexit\n\truntime/asm_amd64.s:1357\nfailed to update info"]

GanttDue: 2020-04-03

**Comment by amyangfei** [comment]
some key logs

```
[2020/03/27 03:08:34.930 +00:00] [INFO] [owner.go:1153] ["mark processor down"] [processorid=d9c7d9fc-db9b-4682-bcf5-d34862d3a4fb] [captureid=dd1d353c-6d54-40ea-b26e-ed3652c0dcbd
] [changefeed=285fec9d-8e8c-4dde-a7d2-145901b4ded7] [tables="[{\"id\":47,\"start-ts\":415571039479136259},{\"id\":57,\"start-ts\":415571039479136259},{\"id\":61,\"start-ts\":4155
71039479136259}]"]
[2020/03/27 03:08:34.966 +00:00] [DEBUG] [owner.go:444] ["readd table"] [tid=47] [startts=415571039479136259]
[2020/03/27 03:08:34.966 +00:00] [DEBUG] [owner.go:444] ["readd table"] [tid=57] [startts=415571039479136259]
[2020/03/27 03:08:34.966 +00:00] [DEBUG] [owner.go:444] ["readd table"] [tid=61] [startts=415571039479136259]
[2020/03/27 03:08:34.966 +00:00] [DEBUG] [owner.go:448] ["delete task status for down processor"] [captureid=dd1d353c-6d54-40ea-b26e-ed3652c0dcbd]
[2020/03/27 03:08:34.974 +00:00] [INFO] [owner.go:479] ["delete capture info"] [id=dd1d353c-6d54-40ea-b26e-ed3652c0dcbd]
```

```
cdc0(2).log:[2020/03/27 03:08:35.837 +00:00] [DEBUG] [processor.go:578] ["Add table"] [tableID=57]
cdc1(2).log:[2020/03/27 03:08:35.975 +00:00] [DEBUG] [processor.go:578] ["Add table"] [tableID=47]
cdc1(2).log:[2020/03/27 03:08:35.976 +00:00] [DEBUG] [processor.go:578] ["Add table"] [tableID=61]
```

cdc0
```
[2020/03/27 03:46:27.735 +00:00] [DEBUG] [owner.go:833] ["wait checkpoint ts"] [cid=8dbac56b-5385-4873-b250-3f7fa81f2e8f] ["checkpoint ts"=415571039374278663] ["finish ts"=415571039479136259] ["ddl query"="CREATE TABLE hfzny (id int(11) NOT NULL AUTO_INCREMENT,nzpafzn int(17) NULL,wtivuzv int(11) NOT NULL DEFAULT 877709034,sdpvdnni int(19),nkugdzfd varchar(989) NULL,foaivo datetime NULL,gfuyh int(14) NOT NULL DEFAULT 1436600620,lelnbulka int(10) NOT NULL DEFAULT 414726345,gphmq timestamp DEFAULT '2027-12-15 15:19:10',uuid varchar(253),kgsvl int(8),qmpwnsqyh int(11) NULL DEFAULT 1680658502,iratiki text,agfdkf int(10),cawtdv datetime DEFAULT '2033-05-01 03:47:45',PRIMARY KEY(id)) ENGINE = InnoDB DEFAULT CHARACTER SET = utf8"]
[2020/03/27 03:46:27.835 +00:00] [DEBUG] [processor.go:353] ["update task position"] [status="{\"checkpoint-ts\":415571039374278663,\"resolved-ts\":415571054067449857}"]
```

cdc1
```
[2020/03/27 03:47:08.973 +00:00] [DEBUG] [processor.go:353] ["update task position"] [status="{\"checkpoint-ts\":415571039479136259,\"resolved-ts\":415571857040736257}"]
[2020/03/27 03:47:09.473 +00:00] [DEBUG] [processor.go:353] ["update task position"] [status="{\"checkpoint-ts\":415571039479136259,\"resolved-ts\":415571857132486657}"]
[2020/03/27 03:47:09.973 +00:00] [DEBUG] [processor.go:353] ["update task position"] [status="{\"checkpoint-ts\":415571039479136259,\"resolved-ts\":415571857302880257}"]
```

**Comment by amyangfei** [comment]
we have refactored the HA module in https://github.com/pingcap/ticdc/pull/371 and https://github.com/pingcap/ticdc/pull/358
